src/utils/helpers.ts

```diff
@@ -139,6 +139,53 @@ async function getDiscussionUrl(
   return url;
 }
 
+export async function getDiscussionState(
+  notification: Notification,
+  token: string,
+): Promise<DiscussionStateType> {
+  const response: GraphQLSearch<DiscussionStateSearchResultEdge> =
+    await apiRequestAuth(`https://api.github.com/graphql`, 'POST', token, {
+      query: `{
+        search(query:"${formatSearchQueryString(
+          notification.repository.full_name,
+          notification.subject.title,
+          notification.updated_at,
+        )}", type: DISCUSSION, first: 10) {
+          edges {
+            node {
+              ... on Discussion {
+                viewerSubscription
+                title
+                stateReason  
+                isAnswered
+              }
+            }
+          }
+        }
+      }`,
+    });
+  let edges =
```

Comment:
I'm missing context here. This looks like it's doing manual filtering. Why? Is there no better way? What are the edge cases and/or tradeoffs? Should those answers be in the code somehow?

Reply:
I agree there may be an opportunity to simplify the GraphQL response parsing.  For now, I've kept it as similar to the Discussion Comment URL lookup logic.  

I'd like to do a few things with the GraphQL-based queries in separate PRs
* #823
* use GraphQL query variables, instead of string building (susceptible to injection issues)
* bypass querying for `edges` -> `node` and use the shortcut convention `nodes`